platform/iphone/gl_view.mm

```diff
@@ -357,15 +174,41 @@ - (BOOL)createFramebuffer {
 		vm.width = backingWidth;
 		vm.height = backingHeight;
 		vm.resizable = false;
-		OS::get_singleton()->set_video_mode(vm);
+		OS::get_singleton()->set_video_mode(OSIPhone::get_singleton()->get_video_mode());
 		OSIPhone::get_singleton()->set_base_framebuffer(viewFramebuffer);
-	};
+	}
+
+	// Save the gl reference to the frame buffer
 	gl_view_base_fb = viewFramebuffer;
 
 	return YES;
 }
 
-// Clean up any buffers we have allocated.
+- (void)createRenderBuffer {
+
+	// Generate a gl render buffer and make it active
+	glGenRenderbuffersOES(1, &viewRenderbuffer);
+	glBindRenderbufferOES(GL_RENDERBUFFER_OES, viewRenderbuffer);
+
+	// This call associates the storage for the current render buffer with the EAGLDrawable (our CAEAGLLayer)
+	// allowing us to draw into a buffer that will later be rendered to screen wherever the layer is (which corresponds with our view).
+	[context renderbufferStorage:GL_RENDERBUFFER_OES
+					fromDrawable:(id<EAGLDrawable>)self.layer];
+	glFramebufferRenderbufferOES(GL_FRAMEBUFFER_OES, GL_COLOR_ATTACHMENT0_OES, GL_RENDERBUFFER_OES, viewRenderbuffer);
```

Comment:
Might be worth investigating the options here. I know that *all* OpenGL running on iOS today is actually running through a Metal compatibility layer.

That's a PR for another time, but I'd love to build a Metal driver for the visual server to function on iOS and OS X.